Reference material for Aliengrep

mjambon opened the pull request:
The main goal of this document is to let adventurous users experiment with these new features and collect their feedback so we can make adjustments.

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

adamkvitek reviewed (approved):
@mjambon thank you for this reference document, before we merge it, can we remove or work on the three remaining TODOs you left in the document? I pushed some small edits that did not change the meaning of what you originally wrote. I also included Aliengrep to sidenav...

Just listing the small edits I did:
- I tried to use more active voice instead of passive voice when possible.
- I removed future tense for easier reading (standard of simple English in docs - use future tense only when referring to a specific time in the future).
- I formatted the paragraphs into one-liners. I know that in code soft-wrapping is not welcomed much, but it is beneficial for docs as it is much easier to edit paragraphs that are not split after 80 (or any number of) characters.
- I made small edits such as more consistent use of `target input` instead of just using `target` or `input` (but I left only `input` in a few occurrences where it can be too repetitive).

mjambon commented:
Thanks @adamkvitek.
* I removed one TODO that needed to be removed and converted the two other ones into "note" admonitions. They're calls for feedback and it's the best I could find (list of Docusaurus admonitions: https://docusaurus.io/docs/markdown-features/admonitions).
* I used vscode instead of emacs so as to not break the lines.
